dali/kernels/kernel_req.h

```diff
@@ -25,13 +25,31 @@
 namespace dali {
 namespace kernels {
 
+using scratch_sizes_t = std::array<size_t, static_cast<size_t>(AllocType::Count)>;
+
+inline scratch_sizes_t GetMaxScratch(const scratch_sizes_t &a, const scratch_sizes_t &b) {
+  scratch_sizes_t result;
+  for (size_t i = 0; i < result.size(); i++) {
+    result[i] = std::max(a[i], b[i]);
+  }
+  return result;
+}
+
+inline scratch_sizes_t GetSumScratch(const scratch_sizes_t &a, const scratch_sizes_t &b) {
```

Comment:
```suggestion
inline scratch_sizes_t AppendScratchSize(const scratch_sizes_t &a, const scratch_sizes_t &b, int alignment = 64) {
```

Reply:
Also, use this function in AddInputSet - the existing implementation is buggy.

Reply:
done